crates/orchestrator/src/lib.rs

```diff
@@ -328,7 +334,21 @@ where
 
         let node_index = self.pub_posted.len() as u64;
 
+        // Deserialize the public key
         let staked_pubkey = PeerConfig::<KEY>::from_bytes(pubkey).unwrap();
+
+        // Check if the node is allowed to connect
+        if !self
+            .config
+            .public_keys
+            .contains(&staked_pubkey.stake_table_entry.public_key())
```

Comment:
I think this should be `if !self.config.disable_verification && !self.config.public_keys.contains(..)`

(though the double negatives are confusing, maybe it should be `enable_verification`? that's very minor though)